doc/dune-files.rst

```diff
@@ -603,6 +603,10 @@ binary at the same place as where ``ocamlc`` was found.
 Executables can also be linked as object or shared object files. See
 `linking modes`_ for more information.
 
+Starting from dune 2.8, executable modules without interface files (`.mli` or
+`.rei`) are compiled against empty interfaces in order to improve dead code
+detection.
```

Comment:
I believe the feature works with any dialect defined using a `(dialect ...)` stanza, not just for `.mli` or `.rei` files.

Reply:
Indeed, thanks. Changed to "(e.g. `.mli`, `.rei`)".